terraform/permissions/main.tf

```diff
@@ -0,0 +1,251 @@
+  bucket = var.berglas_bucket
+  object = berglas_secret.media_bucket.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.media_bucket]
+}
+
+
+resource google_storage_object_access_control media_bucket_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.media_bucket.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.media_bucket, google_storage_object_access_control.media_bucket_sa_email]
+}
+
+
+#
+# Defining secret "superpass"
+resource berglas_secret superpass {
+  bucket    = var.berglas_bucket
+  key       = local.berglas_key
+  name      = "superpass"
+  plaintext = local.berglas_secrets_list.superpass
+}
+
+
+resource google_storage_object_access_control superpass_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.superpass.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.superpass]
+}
+
+
+resource google_storage_object_access_control superpass_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.superpass.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.superpass, google_storage_object_access_control.superpass_sa_email]
+}
+
+
+#
+# Defining secret "database_url"
+resource berglas_secret database_url {
+  bucket    = var.berglas_bucket
+  key       = local.berglas_key
+  name      = "database_url"
+  plaintext = local.berglas_secrets_list.database_url
+}
+
+
+resource google_storage_object_access_control database_url_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.database_url.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.database_url]
+}
+
+
+resource google_storage_object_access_control database_url_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.database_url.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.database_url, google_storage_object_access_control.database_url_sa_email]
+}
+
+
+#
+# Defining secret "secret_key"
+resource berglas_secret secret_key {
+  bucket    = var.berglas_bucket
+  key       = local.berglas_key
+  name      = "secret_key"
+  plaintext = local.berglas_secrets_list.secret_key
+}
+
+
+resource google_storage_object_access_control secret_key_sa_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.secret_key.name
+  role   = "READER"
+  entity = "user-${local.sa_email}"
+
+  depends_on = [berglas_secret.secret_key]
+}
+
+
+resource google_storage_object_access_control secret_key_sa_cb_email {
+  bucket = var.berglas_bucket
+  object = berglas_secret.secret_key.name
+  role   = "READER"
+  entity = "user-${local.sa_cb_email}"
+
+  depends_on = [berglas_secret.secret_key, google_storage_object_access_control.secret_key_sa_email]
+}
+
+# GENERATED SECTION END
```

Comment:
For readability, I'd put the generated stuff in it's own file like `foo_gen.tf`. Terraform reads all the configs once, so locals can be defined in another file.

Reply:
Thanks! I really appreciate the terraform insights